### DISCUSSION TITLE
clarification on command blocks in v0.8

### DISCUSSION THREAD

**Discussion by reformman1001** [wavetermdev/waveterm#1084] [General]
### Current Behavior

It looks like blocks for each executed command,  imho one of the defining features of Wave, is gone after 8.X. Looks like terminal windows are still called "blocks", but they really seem to be just windows. You can't relaunch them, they aren't created automatically with each command, you don't get info about execution etc.

Is this intended? If so, are there any plans to bring them back, or are they gone forever? 

Without this, it's pretty much like any other terminal emulator out there :/ 

### Expected Behavior

An update of what it was like at 7.6 or before, not removal

### Steps To Reproduce

-

### Wave Version

0.8.12

### Architecture

None

### Questionnaire

- [ ] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by esimkowitz** [comment]
Converting to a discussion. Sorry for the confusion about v0.7 vs v0.8, I understand your frustration. The redesign was not something we took lightly and I too am bummed about having to drop some of the features we had in the old UI. I’m optimistic that we will be able to bring a lot of the features from the old app back to the new one in a way that won’t be as disruptive to people’s existing workflows. Workspaces will be making a return in v0.9 and we’re planning to bring back restartable single command blocks very soon. The block-style command history will take longer, unfortunately, as our underlying architecture is very different, but we’ve started brainstorming how to bring it back

We have rereleased the old app as WaveLegacy, which you can download here: https://www.waveterm.dev/download-legacy

**Comment by reformman1001** [comment]
I see, thanks for responding. Guess I'll have to stay on legacy or move on to something else if you take a different direction after all...

It's unfortunate, as the feature has been a game changer for me. For example, for the last 2 weeks I had to build many giant projects I haven't worked with before, and then execute multiple other tools on build results. All of this included many long commands I would normally have to copy&paste, take notes, try to look them up again in large piles of output etc. A process I despise deeply. Meanwhile with this new workflow, I have to figure it out once, remove what isn't working and then it's forever there. I can just execute it again, without ever asking myself "what was that 3rd param of that tool?". 

I'd be willing to pay a subscription for this on my own if my company refused to buy it for me. Hell, I would even buy Warp if they had a policy my company would be ok with.

**Reply by reformman1001** [reply]
For anyone else interested in this: doesn't look like this is coming soon (or maybe at all): 
https://github.com/wavetermdev/waveterm/discussions/989#discussioncomment-10882560

**Reply by ljak** [reply]
Sad, and a bummer. Block based commands was the killer feature for me, the only reason I gave it a try to Wave. The rest of the features is like...any other terminals. 
As today, I only wish somebody could fork the "legacy" Wave...

**Reply by stoo-cloud** [reply]
I came to wave for these blocks and was greeted with 0.8.13 and thought something was wrong and had to try work out what has changed. It is a shame, I really liked what I saw of the old design on a lengthy youtube video on the wave channel that seems to be for version 0.5.3

**Reply by esimkowitz** [reply]
The legacy Wave still exists and the source code can be found in the `main-legacy` branch. You can download it [here](https://www.waveterm.dev/download-legacy). We’ve had some internal discussions this week about how we could bring back the block style command output in a way that won’t break as much. I think if we were to bring it back, it would be optional, so if you notice issues you could disable it and get the raw terminal output. There’s still a good deal of unknowns and I have no firm deadline, but I promise work is underway to bring back as many features as we can, while aligning to our new vision. Stay tuned, we should have more to share soon!

**Reply by reformman1001** [reply]
Thanks, fingers crossed! 

Where can I read more about the new vision for the project? I assume the goal now is even heavier focus on AI features?

**Reply by esimkowitz** [reply]
> Thanks, fingers crossed! 
> 
> 
> 
> Where can I read more about the new vision for the project? I assume the goal now is even heavier focus on AI features? 

There's certainly AI features we're working on but it's not our only focus. We're meeting up in person this week for GitHub Universe so hopefully will have more to share after!

**Reply by texastoland** [reply]
I'm finally switching back to Warp. I had switched to Wave Legacy for a while because it felt more polished. I do miss the blocks, but the new UI feels off in general. For example, the panel animation uses linear easing instead of a more natural function. I don't think switching to web technology is inherently bad, but it's hard to love compared to the unmaintained legacy app. I appreciate that it's a passion project and that it must have been a difficult decision. I'm just sharing my feedback after going back and forth for a few months.

**Comment by felixsanz** [comment]
> Without this, it's pretty much like any other terminal emulator out there :/

not only that... this is now just a terminal inside electron wrapper. so bad... sadly uninstalled it :( Hope 0.9 bring them back but to be honest I don't like the direction this is taking. I don't need a file manager, a web browser or a "sysinfo" panel... there are better tools for that. this is kinda non-sense tbh

**Comment by chastain** [comment]
I also downloaded Wave because of an older YT video highlighting the block based command feature but was disappointed to find the feature (still) missing from 0.11. Has a decision been made to bring it back or not?

**Comment by felipelazzari** [comment]
I also think the input output blocks were the killer feature of this terminal, any updates on bringing it back?

**Comment by mikenerone** [comment]
Spent the morning under the assumption that I'm switching to Wave based on an outdated Youtube video. I immediately saw that there were no blocks, but made the bad assumption that there was some config I'd need to set, and intended to go figure that out later since there's no Settings UI (!). When I finally got around to it, I discovered that blocks are just gone, and this discussion. I'm very disappointed. Probably back to Warp I go.
